app/models/ems_event.rb

```diff
@@ -146,6 +146,34 @@ def self.process_vm_in_event!(event, options = {})
         event[:vm_name] ||= vm.name
       end
     end
+
+    event_type = event[:event_type]
+    # both event types are Rhev specific
+    if (event_type == 'USER_ADD_VM' || event_type == 'USER_ADD_VM_FINISHED_SUCCESS') && event[:vm_or_template_id].nil?
+      data = event[:full_data]
+      vm = data[:vm]
+      cluster = data[:cluster]
+
+      ems_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(vm[:href])
+      cluster_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(cluster[:href])
+
+      vm_data = [ems_ref.include?('/templates/'), ems_ref, vm[:id], event[:message].split(/\W/)[1]]
+      vm_hash = ManageIQ::Providers::Redhat::InfraManager::RefreshParser.create_vm_hash(vm_data)
+
+      old_cluster = EmsCluster.find_by(:ems_ref => cluster_ref)
+      vm_hash[:ems_cluster_id] = old_cluster[:id]
+
+      ems = ExtManagementSystem.find_by_id(event[:ems_id])
+      new_vm = ems.vms_and_templates.build(vm_hash)
+      new_vm.save!
```

Comment:
@blomquisg Could you or someone on the provider team review this?  The idea is to create a minimal VM object based on the event so a targeted refresh can be used instead of refreshing the whole EMS.

I am pretty sure this is the first time someone is trying to approach and concerned it would expose us to the multiple parents bug mentioned here https://github.com/ManageIQ/manageiq/pull/10176#issuecomment-237906310

cc @Fryguy